src/transformers/pytorch_utils.py

```diff
@@ -285,4 +285,14 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
     guaranteed to be unique and constant for this tensor's storage during its lifetime. Two tensor storages with
     non-overlapping lifetimes may have the same id.
     """
+    if tensor.device.type == 'xla':
```

Comment:
This should check if torch_xla is installed:
```suggestion
    if tensor.device.type == 'xla' and is_torch_tpu_available():
```

Please also ensure you add `from .utils import is_torch_tpu_available` to the top of the file

Reply:
done.